contracts/interchain-token-service/tests/canonical_token.rs

```diff
@@ -0,0 +1,57 @@
+mod utils;
+
+use axelar_soroban_std::{address::AddressExt, assert_contract_err, events};
+use interchain_token_service::{
+    error::ContractError, event::InterchainTokenIdClaimed, types::TokenManagerType,
+};
+use soroban_sdk::{testutils::Address as _, xdr::ToXdr, Address, BytesN};
+use utils::setup_env;
+
+const PREFIX_CANONICAL_TOKEN_SALT: &str = "canonical-token-salt";
+
+#[test]
+fn register_canonical_token_succeeds() {
+    let (env, client, _, _) = setup_env();
+    let token_address = Address::generate(&env);
+
+    let chain_name = client.chain_name();
+    let chain_name_hash: BytesN<32> = env.crypto().keccak256(&(chain_name).to_xdr(&env)).into();
+    let expected_deploy_salt = env
+        .crypto()
+        .keccak256(
+            &(
+                PREFIX_CANONICAL_TOKEN_SALT,
+                chain_name_hash,
+                token_address.clone(),
+            )
+                .to_xdr(&env),
+        )
+        .into();
+    let expected_id = client.interchain_token_id(&Address::zero(&env), &expected_deploy_salt);
```

Comment:
use goldie to test the token_id derivation matches. Basically, `goldie` can be used to create expectations that a certain computed data doesn't change. We can use this for events, token ids, salts, hashes etc.

goldie can save one file per test, so you can create a separate test to check token_id is derived correctly.

Another option is to do something like:
https://github.com/axelarnetwork/axelar-amplifier/blob/fa798314a2035aeeba06cd4905ae4a9ed71b27b8/contracts/interchain-token-service/tests/execute.rs#L366

But I prefer the former here since they're distinct tests